common/src/autogluon/common/model_filter/model_filter.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from typing import Any, Dict, List, Optional, Union
+
+logger = logging.getLogger(__name__)
+
+
+class ModelFilter:
+    """Class to filter models given user requirements"""
+
+    @staticmethod
+    def include_models(
+        models: Union[Dict[str, Any], List[str]], included_model_types: List[str]
```

Comment:
Why support both list and dict input for `models`? When is list ever used? Ditto for `exclude_models`

Reply:
I had a look at TimeSeries implementation of `excluded_model_types` and it was using list. We could of course change how TimeSeries is implementing the logic but I feel it could be a valid use case while not adding too much difficulty in maintaining

Reply:
+1 on list support: `include=['RF', 'NN']` is a legit use case.